File: VB/DataLayer/Data/Classes/Transaction.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.ComponentModel.DataAnnotations.Schema

Namespace DataLayer.Data.Classes
    Public Class Transaction
        <DatabaseGenerated(DatabaseGeneratedOption.Identity)> _
        Public Property ID() As Integer
        Public Overridable Property Account() As Account
        Public Overridable Property Category() As Category
        <NotMapped> _
        Public Property TransactionType() As TransactionType
            Get
                Return CType(TransactionTypeId, TransactionType)
            End Get
            Set(ByVal value As TransactionType)
                TransactionTypeId = CInt(value)
            End Set
        End Property
        Public Property TransactionTypeId() As Integer
        Public Property TrasactionDate() As Date
        Public Property Amount() As Decimal
        Public Property Comment() As String
    End Class
End Namespace

File: CS/DataLayer/Data/Classes/Transaction.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.ComponentModel.DataAnnotations.Schema;

namespace DataLayer.Data.Classes
{
    public class Transaction {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int ID { get; set; }
        public virtual Account Account { get; set; }
        public virtual Category Category { get; set; }
        [NotMapped]
        public TransactionType TransactionType { get { return (TransactionType)TransactionTypeId; } set { TransactionTypeId = (int)value; } }
        public int TransactionTypeId { get; set; }
        public DateTime TrasactionDate { get; set; }
        public decimal Amount { get; set; }
        public string Comment { get; set; }
    }
}
